dvc/render/data.py

```diff
@@ -0,0 +1,204 @@
+from copy import deepcopy
+from functools import partial
+from typing import Dict, List, Optional, Set, Union
+
+from funcy import first, project, reraise
+
+from dvc.exceptions import DvcException
+from dvc.render.base import INDEX_FIELD, REVISION_FIELD
+
+
+class FieldsNotFoundError(DvcException):
+    def __init__(self, expected_fields, found_fields):
+        self.expected_fields = expected_fields
```

Comment:
Excuse me, what does these two member variable do? And what is the status of this PR?

Reply:
I was using them to reraise in some previous version of this change. Removing.
As of status, I am currently trying to check why benchmarks keep failing. Besides that I believe its ready to merge.

Reply:
The PR is ready for review. @karajan1001 ping me if you want to review it once again. If not, I will squash the minor commits and merge.